core/src/main/java/hudson/PluginFirstClassLoader2.java

```diff
@@ -0,0 +1,167 @@
+     *       <p>Invoke {@link #getResource(String)} on the system class loader.
+     *   <li>
+     *       <p>Invoke {@link #findResource(String)} to find the resource.
+     *   <li>
+     *       <p>Invoke {@link #getResource(String)} on the parent class loader.
+     * </ol>
+     *
+     * @param name The resource name
+     * @return {@link URL} object for reading the resource; {@code null} if the resource could not
+     *     be found, a {@link URL} could not be constructed to locate the resource, the resource is
+     *     in a package that is not opened unconditionally, or access to the resource is denied by
+     *     the security manager.
+     * @throws NullPointerException If {@code name} is {@code null}
+     */
+    @Override
+    public URL getResource(String name) {
+        Objects.requireNonNull(name);
+        URL url;
+        if (systemClassLoader != null) {
+            url = systemClassLoader.getResource(name);
+        } else {
+            url = null;
+        }
+        if (url == null) {
+            url = findResource(name);
+        }
+        if (url == null) {
+            url = getParent().getResource(name);
+        }
+        return url;
+    }
+
+    /**
+     * Find all the resources with the given name. A resource is some data (images, audio, text,
+     * etc) that can be accessed by class code in a way that is independent of the location of the
+     * code.
+     *
+     * <p>The name of a resource is a {@code /}-separated path name that identifies the resource.
+     * This method first invokes {@link #getResources(String)} on the system class loader (if
+     * available). It then invokes {@link #findResources(String)} to find the resources with the
+     * name in this class loader. Finally, it invokes {@link #getResources(String)} on the parent
+     * class loader. It returns an enumeration whose elements are the {@link URL}s found by
+     * searching the system class loader (if available), followed by the {@link URL}s found with
+     * {@link #findResources(String)}, followed by the {@link URL}s found by searching the parent
+     * class loader.
+     *
+     * @param name The resource name
+     * @return An enumeration of {@link URL} objects for the resource. If no resources could be
+     *     found, the enumeration will be empty. Resources for which a {@link URL} cannot be
+     *     constructed, which are in a package that is not opened unconditionally, or for which
+     *     access to the resource is denied by the security manager, are not returned in the
+     *     enumeration.
+     * @throws IOException If I/O errors occur
+     * @throws NullPointerException If {@code name} is {@code null}
+     */
+    @Override
+    public Enumeration<URL> getResources(String name) throws IOException {
+        Objects.requireNonNull(name);
+        if (systemClassLoader != null) {
+            return new CompoundEnumeration<>(
+                    systemClassLoader.getResources(name),
```

Comment:
ditto the enumeration order with system classloader.

Reply:
I removed any references to the system class loader.